packages/server/tests/AuthenticationSignUp.test.ts

```diff
@@ -230,7 +253,55 @@ test.group('Authentication sign up tests group', (group) => {
             } as SignUpRequestBody)
             .expect(400);
 
-        const { status } = SignUpResponseBody.parse(rawBody);
-        assert.equal(status, 'UNAVAILABLE_EMAIL');
+        const { status, signFailureReasons } =
+            SignUpFailureResponseBody.parse(rawBody);
+        assert.equal(status, 'FAILURE');
+        assert.isDefined(signFailureReasons);
+        assert.isDefined(
+            signFailureReasons.find((error) => error === 'UNAVAILABLE_EMAIL'),
+        );
+    });
+
+    test('It should fail to sign up raising every possibles errors', async (assert) => {
+        const email = internet.email().replace('@', random.word());
+        const userNickname = internet.userName();
+
+        // Indeed we're saving a bad formatted email directly in database
+        // To cover corresponding exception for the following sign up
+        await User.create({
+            email,
+            nickname: userNickname,
+            password: 'nevermind',
+            uuid: datatype.uuid(),
+        });
+
+        const request = supertest.agent(BASE_URL);
+        const password = generateWeakPassword();
+
+        const { body: rawBody } = await request
+            .post(urlcat(TEST_AUTHENTICATION_GROUP_PREFIX, 'sign-up'))
+            .send({
+                authenticationMode: 'api',
+                email,
+                password,
+                userNickname,
+            } as SignUpRequestBody)
+            .expect(400);
+
+        const { status, signFailureReasons } =
+            SignUpFailureResponseBody.parse(rawBody);
+        assert.equal(status, 'FAILURE');
+        assert.isDefined(signFailureReasons);
+        assert.equal(signFailureReasons.length, 4);
+        [
+            'UNAVAILABLE_NICKNAME',
+            'UNAVAILABLE_EMAIL',
+            'INVALID_EMAIL',
+            'WEAK_PASSWORD',
+        ].forEach((reasonValue) =>
+            assert.isDefined(
+                signFailureReasons.find((error) => error === reasonValue),
+            ),
+        );
```

Comment:
**<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**🎯 suggestion:** This is a matter of coding style, but this could be written in a more concise way, like so:

```suggestion
        assert.deepEqual(signFailureReasons, ['UNAVAILABLE_NICKNAME', 'UNAVAILABLE_EMAIL', 'INVALID_EMAIL', 'WEAK_PASSWORD' ]);
```